src/Bicep.Core/Semantics/Namespaces/AzNamespaceType.cs

```diff
@@ -177,6 +177,10 @@ private static ObjectType GetDeploymentReturnType(ResourceScope targetScope)
                 new TypeProperty("name", LanguageConstants.String),
                 new TypeProperty("properties", new ObjectType("properties", TypeSymbolValidationFlags.Default, new []
                 {
+                    new TypeProperty("template", new ObjectType("properties", TypeSymbolValidationFlags.Default, new []
```

Comment:
Although for templateLink is this same case, I think it would be better if object name in ObjecType constructor matched the property name.

Reply:
Good point! We could rename this one "properties" -> "TemplateProperties", and the other (line 184) "properties" -> "TemplateLinkProperties". @slapointe would you mind fixing both?

Reply:
just keep the camelCase style: `templateProperties`, `templateLinkProperties` 😉

Reply:
Got it.

There are a few occurences elsewhere in the same file like that. Do you want me to fix them all?

```c#
private static ObjectType GetEnvironmentReturnType()
        {
            return new ObjectType("environment", TypeSymbolValidationFlags.Default, new[]
            {
                new TypeProperty("activeDirectoryDataLake", LanguageConstants.String),
                new TypeProperty("authentication", new ObjectType("authentication", TypeSymbolValidationFlags.Default, new []
                {
                    new TypeProperty("audiences", new TypedArrayType(LanguageConstants.String, TypeSymbolValidationFlags.Default)),
                    new TypeProperty("identityProvider", LanguageConstants.String),
                    new TypeProperty("loginEndpoint", LanguageConstants.String),
                    new TypeProperty("tenant", LanguageConstants.String),
                }, null)),
                new TypeProperty("batch", LanguageConstants.String),
                new TypeProperty("gallery", LanguageConstants.String),
                new TypeProperty("graph", LanguageConstants.String),
                new TypeProperty("graphAudience", LanguageConstants.String),
                new TypeProperty("media", LanguageConstants.String),
                new TypeProperty("name", LanguageConstants.String),
                new TypeProperty("portal", LanguageConstants.String),
                new TypeProperty("resourceManager", LanguageConstants.String),
                new TypeProperty("sqlManagement", LanguageConstants.String),
                new TypeProperty("suffixes", new ObjectType("suffixes", TypeSymbolValidationFlags.Default, new []
                {
                    new TypeProperty("acrLoginServer", LanguageConstants.String),
                    new TypeProperty("azureDatalakeAnalyticsCatalogAndJob", LanguageConstants.String),
                    new TypeProperty("azureDatalakeStoreFileSystem", LanguageConstants.String),
                    new TypeProperty("azureFrontDoorEndpointSuffix", LanguageConstants.String),
                    new TypeProperty("keyvaultDns", LanguageConstants.String),
                    new TypeProperty("sqlServerHostname", LanguageConstants.String),
                    new TypeProperty("storage", LanguageConstants.String),
                }, null)),
                new TypeProperty("vmImageAliasDoc", LanguageConstants.String),
            }, null);
        }
```
In this example `suffixes` & `authentication` in `GetEnvironmentReturnType`

Reply:
That would be great!

Reply:
Done, but my merge pulled a lot of other files too. Not intended, sorry.

Reply:
Done, let me know if that's good with you.